### PR TITLE
fixed the database

### PR DESCRIPTION
**Pull request by RezaAmraei**
database stuff

### CODE REVIEW

**Comment by RezaAmraei** [comment]
feat : added route and controller for login

File: server/controllers.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
+const db = require("../DataBase/index");
+const bcrypt = require("bcrypt");
+const { query } = require("express");
```
Comment (ViniBjazevich) [inline server/controllers.js]: ??????

File: server/controllers.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  let { first_name, last_name, email, password, username, bio, location } =
+    req.body;
+
+  bcrypt.hash(password, 10).then((hash) => {
+    password = hash;
+    db.query(
+      `INSERT INTO accounts (first_name, last_name , username, password, email, bio, location) VALUES ('${first_name}','${last_name}','${email}','${password}','${username}','${bio}','${location}');`
+    )
+      .then(() => res.status(200).send("account created"))
+      .catch((err) => console.log("err"));
```
Comment (ViniBjazevich) [inline server/controllers.js]: send 400 on error

File: server/controllers.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    if (!passwordCheck) {
+      console.log("passwords dont match");
+      return res.status(400);
+    }
+    console.log("hello correct user");
+    res.json({ id: user.id });
+    //checks to see if it is the correct password
+  } catch (e) {
+    console.error(e.stack);
+    res.send(e.stack);
```
Comment (ViniBjazevich) [inline server/controllers.js]: Only send 400 if call errors out

File: server/controllers.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    let user = results.rows[0];
+    console.log(user);
+    //checks to see if there is a user with that email
+    if (!user) {
+      return res.status(400);
+    }
+    const passwordCheck = await bcrypt.compare(password, user.password);
+    if (!passwordCheck) {
+      console.log("passwords dont match");
+      return res.status(400);
```
Comment (ViniBjazevich) [inline server/controllers.js]: Return true if password is correct. Else return false

**Comment by RezaAmraei** [comment]
chore - got ride of comments in controller and added 400 status in the catch

**Comment by RezaAmraei** [comment]
feat : added registration through the frontend, still needs validations to be added

**Pull request merged**